README.md

```diff
@@ -1 +1,145 @@
-# service kit
+# Service Kit Documentation
+
+![License](https://img.shields.io/github/license/guardicode/service-kit)
+
+
+ 📌 Service-Kit is a collection of functions and objects designed to help the
+Monkey team bootstrap, build, and maintain various services efficiently.
+
+## Features
+
+- **Service Bootstrapping**: Provides templates and utilities to initialize
+new services quickly.
+- **Common Utilities**: Provides common utilities for logging, configuration,
+and error handling.
+- **Service Configuration**: Provides a base class for service
+configuration that can be easily extended.
+- **Testing Support**: Provides utilities for testing services.
+
+## Getting started
+
+### Installation
+
+You can install Service-Kit using [poetry](https://python-poetry.org/):
+
+```bash
+poetry add git+https://github.com/guardicode/service-kit.git
+```
+
+or by using pip:
+
+```bash
+pip install git+https://github.com/guardicode/service-kit.git
+```
+
+### Usage
+
+After installation you can start using Service-Kit as any other Python package.
+For more detailed example and usage patterns, refer to the
+`service_template.py` file included in the repository.
+
+### Configuration
+
+Before using Service-Kit you will need to add configuration
+class to your project which will be a subclass of `ServiceConfiguration`.
+This class will define the configuration options for your service.
+Configuration options can be provided either as environment variables or in a
+`.env` file. See
+https://pypi.org/project/python-dotenv/#file-format for more details on `.env`
+file format.
+
+<!--
+DO NOT MODIFY the configuration options documentation below; it is
+autogenerated. To regenerate the documentation, run the
+autogenerate-options-docs.sh script.
+-->
+<!-- generated env. vars. start -->
+#### `BIND_ADDRESS`
+
+*Optional*, default value: `127.0.0.1`
+
+The interface to listen on
+
+#### `DEBUG`
+
+*Optional*, default value: `False`
+
+Enable debug mode (also override the log level)
+
+#### `ENABLE_HOT_RELOAD`
+
+*Optional*, default value: `False`
+
+Enable hot-reloading during development
+
+#### `LOG_DIRECTORY`
+
+*Optional*, default value: `None`
+
+The directory to write log files to (it will be created if it does not exist)
+
+#### `LOG_LEVEL`
+
+*Optional*, default value: `INFO`
+
+The log level to use
+
+##### Possible values
+
+`TRACE`, `DEBUG`, `INFO`, `SUCCESS`, `WARNING`, `ERROR`, `CRITICAL`
+
+#### `PORT`
+
+*Optional*, default value: `8080`
+
+The port to listen on
+
+#### `PRETTY_PRINT_LOGS`
+
+*Optional*, default value: `True`
+
+Enable pretty-printing of JSON logs
+
+#### `SSL_CERTFILE`
+
+*Optional*, default value: `None`
+
+The path to the SSL certificate file
+
+#### `SSL_KEYFILE`
+
+*Optional*, default value: `None`
+
+The path to the SSL key file
+<!-- generated env. vars. end -->
```

Comment:
Do we need this in the README file? Could we put it straight into `configuration.rst`?

Reply:
Instead of generating this using `settings-doc`, does Sphinx have some plugin for generating documentation for `pygantic_settings.BaseSettings` models?